crates/solver/src/liquidity/order_converter.rs

```diff
@@ -71,14 +74,35 @@ struct OrderSettlementHandler {
     scaled_unsubsidized_fee_amount: U256,
 }
 
+/// Returns (`sell_amount`, `fee_amount`) for the given order and adjusts the values accordingly
+/// for limit orders.
+fn compute_synthetic_order_amounts_if_limit_order(order: &Order) -> Result<(U256, U256)> {
+    if order.metadata.class == OrderClass::Liquidity {
+        // adjust amounts to account for `surplus_fee`
+        let sell_amount = order
+            .data
+            .sell_amount
+            .checked_sub(order.metadata.surplus_fee)
+            .context("surplus_fee adjustment would underflow sell_amount")?;
+        let fee_amount = order
+            .data
+            .fee_amount
+            .checked_add(order.metadata.surplus_fee)
+            .context("surplus_fee adjustment would overflow U256 in fee_amount")?;
```

Comment:
I think this is slightly off:
```
let sell_amount = order.data.sell_amount + order.data.fee_amount - order.metadata.surplus_fee;
let fee_amount = order.metadata.surplus_fee;
```

The reasoning is that the synthetic order's `fee_amount` should be exactly the `suplus_fee` for any given auction.

Reply:
Correct, my understanding is that `surplus_fee` is the "new fee" that is defined per batch and is exactly what the fee should be.

Dealing with non-0 `fee_amount` is just for completeness. In practice limit orders have a 0 `fee_amount`.

Reply:
Fixed [5f664cc](https://github.com/cowprotocol/services/pull/713/commits/5f664ccb25b0debaed901367b8f4d8920836c71c) (introduced typo was fixed in [a8bf1e5](https://github.com/cowprotocol/services/pull/713/commits/a8bf1e5c66d85dafac7da458a5ec0bb7c1429abf))